charts/llm-d-modelservice/templates/prefill-podmonitor.yaml

```diff
@@ -19,7 +19,8 @@ spec:
     matchLabels:
       {{- include "llm-d-modelservice.prefilllabels" . | nindent 6 }}
   podMetricsEndpoints:
-  - port: {{ .Values.prefill.monitoring.podmonitor.port | default (include "llm-d-modelservice.vllmPort" (dict "role" "prefill" "Values" .Values)) }}
+  # For prefill, this is port 8000. Must use port name.
+  - port: {{ .Values.prefill.monitoring.podmonitor.port | default "metrics" | quote }}
```

Comment:
What if the user still uses an integer as port? Is the intention for a value like "8080" to not fail the `helm install`?

Reply:
helm will not fail as long as a string is provided, but prometheus confusingly looks for containerPort.name instead of port integer and will fail to create the scrape target from the `port: "8000"` - weird but as far as I can tell from the upstream issues and my testing, true. It is possible to pass `portNumber: 8000` as integer, and I think in this case you will not require to expose the port in the pod. I'll test this and then we'll add this as an option in a follow-up PR since people may want to use this, provided their prometheus version supports the new-ish field `podNumber`.